libdevcore/db.h

```diff
@@ -21,16 +21,80 @@
 
 #pragma once
 
+#include "Exceptions.h"
 #include "dbfwd.h"
 
-#pragma warning(push)
-#pragma warning(disable: 4100 4267)
-#if ETH_ROCKSDB
-#include <rocksdb/db.h>
-#include <rocksdb/write_batch.h>
-#else
-#include <leveldb/db.h>
-#include <leveldb/write_batch.h>
-#endif
-#pragma warning(pop)
-#define DEV_LDB 1
+#include <memory>
+#include <string>
+
+namespace dev
+{
+namespace db
+{
+
+// Transaction implements database transaction for a specific concrete database
+// implementation. The transaction implementation must call rollback in its
+// destructor if commit has not yet been called (making sure to catch any
+// exceptions).
+class Transaction
```

Comment:
Actually, `transaction` has somewhat broader implications than `batch`. A `transaction` is more correct in the case of `LMDB` (to be submitted as separate PR) where the transactions surround read operations as well as write operations.

Reply:
Well this interface has methods only for modification, will it change in LMDB PR ?

Reply:
I guess you are correct that I cannot expose the broader notion because of limitations in the other implementations that only support transactional writes.